src/Zone.js

```diff
@@ -33,23 +30,50 @@ export default class Zone extends UXBase {
 		return mark;
 	});
 
+	// 'state' of the measure that indicates whether it has been recorded
+	measured = false;
+
 	/**
 	 * Records measure on Performance Timeline and calls onMeasure callback
 	 *
 	 * @param {ExpectedMark} lastMark last mark that triggered completion
 	 */
 	measure(endMarkName) {
+		if (this.measured) {
+			// `measure` only allowed to be called once per zone instance
+			return;
```

Comment:
Should we add a console warning for this? Seems like something that shouldn't happen if used right.

Reply:
In SPA apps, I suspect it might actually be pretty common, because the way the Zone is set up, whenever _any_ mark is recorded, _all_ marks are checked. Since we allow marks to have been recorded before a Zone initializes, that means that if _all_ marks have already been recorded, then _each_ mark will result in `.measure` being fired - we only care about the first one.

It parallels the way the ExpectedMarks work - you're allowed to call 'record' on them multiple times, but only the first one has an effect.